python/pyspark/sql/context.pyi

```diff
@@ -63,7 +64,7 @@ class SQLContext:
     ) -> DataFrame: ...
     def registerFunction(
         self, name: str, f: Callable[..., Any], returnType: DataType = ...
-    ) -> None: ...
+    ) -> UserDefinedFunctionLike: ...
```

Comment:
Note for reviewer:
https://github.com/apache/spark/blob/ff1fc5ed4b685b4f5f83d5f600b24f089dd4522e/python/pyspark/sql/context.py#L236
https://github.com/apache/spark/blob/ff1fc5ed4b685b4f5f83d5f600b24f089dd4522e/python/pyspark/sql/udf.pyi#L46-L51